### PR TITLE
feat: add ioc container and font service for BlockSuite

### PR DESCRIPTION
**Pull request by akumatus**
### What Changed?
- Add ioc Container to BlockSuite
- Refactor `FontLoader` to injectable `FontService`

With ioc Container, all dependencies are automatically created. 

In the screen recording below, page mode will not load `CommunityCanvasTextFonts`, but edgeless mode will automatically load `CommunityCanvasTextFonts`. That's because only edgelss mode depends on `FontService`.
<div class='graphite__hidden'>
          <div>🎥 Video uploaded on Graphite:</div>
            <a href="https://app.graphite.dev/media/video/sJGviKxfE3Ap685cl5bj/6a2d9a0e-9fa6-4343-bbc7-1f21ec271eae.mov">
              
            </a>
          </div>
<video src="https://graphite-user-uploaded-assets-prod.s3.amazonaws.com/sJGviKxfE3Ap685cl5bj/6a2d9a0e-9fa6-4343-bbc7-1f21ec271eae.mov">录屏2024-08-16 12.25.00.mov</video>

### CODE REVIEW

**Comment by akumatus** [comment]
* **#7992** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7992?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @akumatus and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/affine/services/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  "name": "@blocksuite/affine-services",
+  "version": "0.16.0",
+  "description": "BlockSuite services for AFFiNE.",
+  "type": "module",
+  "repository": "toeverything/blocksuite",
+  "scripts": {
+    "build": "tsc",
+    "test:unit": "nx vite:test --run --passWithNoTests",
+    "test:unit:coverage": "nx vite:test --run --coverage",
+    "test:e2e": "playwright test"
```
Comment (Saul-Mirone) [inline packages/affine/services/package.json:11]:
```suggestion
    "test": "nx e2e"
```

File: packages/presets/src/ai/utils/custom-specs.ts
[the remark this answers is not in the document]
Comment (akumatus) [reply packages/presets/src/ai/utils/custom-specs.ts:1]: I will remove `presets/ai` and bump BlockSuite to AFFiNE later. The modification here is to show the advantages of ioc, which means we do not need to override fonts loading behavior to different services.

[the remark this answers is not in the document]
Comment (L-Sun) [reply packages/presets/src/ai/utils/custom-specs.ts:1]: LGTM

**Review by L-Sun — APPROVED** [review]

File: packages/framework/block-std/src/scope/block-std-scope.ts [hunk 2]
```diff
@@ -15,6 +17,8 @@ export interface BlockStdOptions {
   doc: Doc;
 }
 
+export const services = new Container();
```
Comment (Saul-Mirone) [inline packages/framework/block-std/src/scope/block-std-scope.ts:20]: According to their readme, seems they need decorator stage 2 and reflect metadata. However we're using decorator stage 3 now.

Comment (Saul-Mirone) [reply packages/framework/block-std/src/scope/block-std-scope.ts:20]: https://github.com/inversify/InversifyJS/issues/1507#issuecomment-2066401389

I think we may need to choose another library.

**Comment by Saul-Mirone** [comment]
It would be great if we can use something like `this.std.container.get(ViewInfo).mode` to get page mode, and `this.std.container.get(ViewInfo).viewport` to get some view-related informations.

**Pull request closed without merging**